pilot/pkg/kube/inject/app_probe.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inject
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"istio.io/istio/pilot/cmd/pilot-agent/status"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	// StatusPortCmdFlagName is the name of the command line flag passed to pilot-agent for sidecar readiness probe.
+	// We reuse it for taking over application's readiness probing as well.
+	// TODO: replace the hardcoded statusPort elsewhere by this variable as much as possible.
+	StatusPortCmdFlagName = "statusPort"
+
+	// TODO: any constant refers to this container's name?
+	istioProxyContainerName = "istio-proxy"
+)
+
+var (
+	statusPortPattern = regexp.MustCompile(fmt.Sprintf("^-{1,2}%s$", StatusPortCmdFlagName))
```

Comment:
Should we support matching  `--statusPort=15020` ?

Reply:
Yes, good catch.